include/boost/geometry/policies/predicate_based_interrupt_policy.hpp

```diff
@@ -43,12 +43,18 @@ struct stateless_predicate_based_interrupt_policy
     template <typename Range>
     inline bool apply(Range const& range)
     {
-        // if there is at least one unacceptable turn in the range, return false
-        has_intersections = !detail::check_iterator_range
-            <
-                IsAcceptableTurnPredicate,
-                AllowEmptyTurnRange
-            >::apply(boost::begin(range), boost::end(range));
+        // if there is at least one unacceptable turn in the range, return true
+        auto const end = boost::end(range);
+        bool const has_unacceptable_turn = std::find_if
+            (
+                boost::begin(range), end,
+                []( auto const& turn ) { 
+                    return ! IsAcceptableTurnPredicate::apply(turn);
+                }
+            ) != end;
```

Comment:
I don't think that this formatting is correct.

First of all we do accept up to 100 characters in a line so you don't have to put everything in subsequent lines.

Another thing is that function name could be placed in the next line (maybe, do we allow this?):
```
bool const has_unacceptable_turn
    = std::find_if(boost::begin(range), end,
                   [](auto const& turn) { 
                       return ! IsAcceptableTurnPredicate::apply(turn);
                   }) != end;
```

Another solution would be to create lambda expression before the call:
```
auto is_unacceptable = [](auto const& turn) { return ! IsAcceptableTurnPredicate::apply(turn); };
```

Reply:
I think I prefer 
```
bool const has_unacceptable_turn = std::find_if(boost::begin(range), end,
    [](auto const& turn) { 
        return ! IsAcceptableTurnPredicate::apply(turn);
    }) != end;
```
what do you think?

Reply:
It looks good to me.